### PR TITLE
[win10] BuildDependencies: update dependencies packages

### PR DESCRIPTION
**Pull request by afedchin**
in order of preparation to #16116

- OpenSSL v1.1.1b
- CURL v7.61.1
- mysql-connector-c -> mariadb-connector-c v3.0.6

### CODE REVIEW

**Comment by candrews** [comment]
Should the dependencies in:
project/BuildDependencies/scripts/0_package.target-win10-arm.list
project/BuildDependencies/scripts/0_package.target-win32.list
project/BuildDependencies/scripts/0_package.target-x64.list

also be updated?

**Comment by afedchin** [comment]
> project/BuildDependencies/scripts/0_package.target-win32.list
> project/BuildDependencies/scripts/0_package.target-x64.list

they were updated already see #14777

> project/BuildDependencies/scripts/0_package.target-win10-arm.list

not supported anymore.

**Comment by candrews** [comment]
> > project/BuildDependencies/scripts/0_package.target-win10-arm.list
> 
> not supported anymore.

So shouldn't it be removed?

**Comment by afedchin** [comment]
> So shouldn't it be removed?

_not supported_ doesn't mean that not working

**Comment by candrews** [comment]
> > project/BuildDependencies/scripts/0_package.target-win32.list
> > project/BuildDependencies/scripts/0_package.target-x64.list
> 
> they were updated already see #14777

Only OpenSSL was updated in that PR; curl is still at 7.59.0 in win32 and 7.61.1 in x64.

I just think it would be nice to get curl in a consistent state, especially while work in this area is being done anyways.

**Comment by afedchin** [comment]
> Only OpenSSL was updated in that PR; curl is still at 7.59.0 in win32 and 7.61.1 in x64.

it's because that only x64 had an issue with curl and ossl, ask @Rechi to sync dependencies for desktop win32
this PR related to uwp only.

**Comment by Rechi** [comment]
The following line also has to be update.
https://github.com/xbmc/xbmc/blob/f712d24821a26ecc522beae58b70cba93103cbcb/tools/buildsteps/windows/make-mingwlibs.sh#L109

Also commit order is wrong, the rebuild check changes should be committed before the dependency updates.

File: tools/buildsteps/windows/make-mingwlibs.sh [hunk 1]
```diff
@@ -73,7 +73,10 @@ checkfiles() {
 
 buildProcess() {
 export PREFIX=/xbmc/project/BuildDependencies/mingwlibs/$TRIPLET
-if [ "$(pathChanged $PREFIX /xbmc/tools/buildsteps/windows /xbmc/tools/depends/target/ffmpeg/FFMPEG-VERSION)" == "0" ]; then
+if [ "$(pathChanged $PREFIX \
+    /xbmc/tools/buildsteps/windows \
+	/xbmc/tools/depends/target/ffmpeg/FFMPEG-VERSION \
+	/xbmc/project/BuildDependencies/scripts)" == "0" ]; then
```
Comment (Rechi) [inline tools/buildsteps/windows/make-mingwlibs.sh]: inconsistent usage of tabs and spaces

**Review by Rechi — APPROVED** [review]
mingwlibs rebuild detection changes are okay

**Comment by Rechi** [comment]
I think you also have to recompile Python, as the hashlib and ssl module depend on `libeay32.dll` and `ssleay32.dll`.

**Comment by afedchin** [comment]
> I think you also have to recompile Python, as the hashlib and ssl module depend on `libeay32.dll` and `ssleay32.dll`.

py3 was compiled with updated dependencies. I see no reason to recompile outdated dependency.

**Comment by candrews** [comment]
curl 7.65.0 was just released addressing 2 CVEs. https://curl.haxx.se/docs/CVE-2019-5435.html may be relevant to Kodi.

In any case, perhaps this PR can be updated before it's merged to use Curl 7.65.0 instead of 7.64.1?

**Comment by MilhouseVH** [comment]
curl 7.65.1 and openssl 1.1.1c now available.

**Comment by fuzzard** [comment]
The dependencies in this have been updated so that part of the PR is obsolete, are any of the changes in Environment.cpp still wanted?

**Comment by phunkyfish** [comment]
@afedchin do you want to progress this, close it or put it on the backburner?

**Comment by phunkyfish** [comment]
As there has been no comment on this PR it will be closed in 2 weeks time if no comment stating to do otherwise is added.

**Comment by phunkyfish** [comment]
Going to close this, can be reopened at a later date if required.

**Pull request closed without merging**